client/solve.go

```diff
@@ -80,27 +80,41 @@ func (c *Client) Solve(ctx context.Context, def *llb.Definition, opt SolveOpt, s
 	switch opt.Exporter {
 	case ExporterLocal:
 		outputDir, ok := opt.ExporterAttrs[exporterLocalOutputDir]
+		delete(opt.ExporterAttrs, exporterLocalOutputDir)
 		if !ok {
 			return errors.Errorf("output directory is required for local exporter")
 		}
 		s.Allow(filesync.NewFSSyncTarget(outputDir))
 	case ExporterOCI, ExporterDocker:
+		outputWriter := FSSyncTargetFileFromContext(ctx)
 		outputFile, ok := opt.ExporterAttrs[exporterOCIDestination]
+		delete(opt.ExporterAttrs, exporterOCIDestination)
 		if ok {
+			if outputWriter != nil {
+				return errors.Errorf("output is already specified via the go context")
+				// TODO(AkihiroSuda): maybe continue with warning? (which do we prefer, ctx or exporter attr here?)
+				// Also, maybe we should consider just removing "output" exporter attr.
```

Comment:
WDYT?

Reply:
> maybe we should consider just removing "output" exporter attr.

Yes, lets pass in the writer directly to the client and move these checks to the `buildctl` or to a public helper function. sgty?

Reply:
These aren't sent to daemon anyway so they are not really exporter attr, other than the cli uses same flags to set them.

Reply:
> lets pass in the writer directly to the client

Do you mean we should use `ClientOpt` (which lifecycle corresponds to the client rather than a solve job) rather than `context` for passing the writer?

> move these checks to the buildctl or to a public helper function

Do we still want to keep `--exporter-opt output` CLI flag?
Or do we want to replace this flag with `--exporter-output`?

Reply:
> Do you mean we should use ClientOpt (which lifecycle corresponds to the client rather than a solve job) rather than context for passing the writer?

Yes

> Do we still want to keep --exporter-opt output CLI flag?
Or do we want to replace this flag with --exporter-output?

Either way works. I'd suggest keeping the current for now.

Reply:
If we use `ClientOpt` for the writer, we can't use reuse the client for different solve jobs, no?

Reply:
Sorry, I didn't mean `ClientOpt`, meant adding it directly to `SolveOpt`.